### ISSUE TITLE
Pylint does not report "unused-import" if there are quotes around type hint

### ISSUE THREAD

**Issue by hofrob** [PyCQA/pylint#7679]
### Bug description

I used [`hatch`](https://hatch.pypa.io) to create this example repo (`pipx install hatch`).

* clone the example repo: https://github.com/hofrob/pylint-unused-import-quotes
* execute `hatch run pylint -d all -e W0611 test_pylint/ tests/`
* pylint should report an unused import in `test_pylint/baz.py`
* remove quotes from type hint in `test_pylint.foo.foo_baz` function (around `bar.SomeClass`)
* execute `hatch run pylint -d all -e W0611 test_pylint/ tests/`
* pylint correctly reports an unused import

Those two things should be unrelated I'd say. No idea why quotes (even if they don't make sense) influence an unused import warning in a completely unrelated file.

### Command used

```shell
pylint -d all -e W0611 test_pylint/ tests/
```

### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

```
************* Module test_pylint.baz
test_pylint/baz.py:1:0: W0611: Unused bar imported from test_pylint (unused-import)

------------------------------------------------------------------
Your code has been rated at 9.17/10 (previous run: 9.17/10, +0.00)
```

### Pylint version

```shell
pylint 2.15.3
astroid 2.12.10
Python 3.10.6 (main, Aug 10 2022, 11:40:04) [GCC 11.3.0]
```

### OS / Environment

* Ubuntu 22.04 
* hatch env `Hatch, version 1.5.0`

**Comment by mbyrnepr2** [comment]
Thanks for the report @hofrob. Unfortunately I can't reproduce it the way you've described (`pylint test_pylint/ tests/`).

However I do see the behaviour when linting individual modules:
```pylint tmp/baz.py tmp/foo.py``` (`import-error` in tmp/baz.py)
```pylint tmp/foo.py tmp/baz.py``` (no `import-error` in tmp/baz.py)
This behaviour is fixed by [this MR](https://github.com/PyCQA/pylint/pull/7621).

**Comment by hofrob** [comment]
Thanks! I will take a look later if it fixes what I'm seeing.

> However I do see the behaviour when linting individual modules:  
> `pylint tmp/baz.py tmp/foo.py` (`import-error` in tmp/baz.py)
> `pylint tmp/foo.py tmp/baz.py` (no `import-error` in tmp/baz.py)

This kind of situation is happening a lot recently. I had a docker container where pylint did not show any messages in our CI, but running the exact same container locally showed the correct messages. I'm running into this a lot at work with co-workers executing pylint with different results. Often times the order of paths in the command line changes the outcome, but it's not the same on all machines (even if they use the same docker container).

Is there any issue regarding this? Do other people report something similar? I couldn't find any threads because I don't even know what to look for to be honest.

**Comment by Pierre-Sassoulas** [comment]
> Is there any issue regarding this? Do other people report something similar? I couldn't find any threads because I don't even know what to look for to be honest.

Yes, there's an issue with multiprocessing and parallelization the hashmap is not implemented on all checkers (https://github.com/PyCQA/pylint/issues/374). Do not use multiprocessing, the increase in performance is negligible anyway (#2525 : the work in astroid to fix this is humongous).

There's also a problem with the order of file given (#689 and #4053)

**Comment by hofrob** [comment]
> This behaviour is fixed by [this MR](https://github.com/PyCQA/pylint/pull/7621).

Can confirm! All false negatives are fixed now.

Thanks for the helpful feedback!
